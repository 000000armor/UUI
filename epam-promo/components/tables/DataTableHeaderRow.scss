@use  '../../assets/styles/index' as *;

.root {
    background: $gray5;
    border-bottom: 1px solid $gray30;

    &:global(.uui-table-header-row .uui-table-fixed-section-left),
    &:global(.uui-table-header-row .uui-table-fixed-section-right) {
        z-index: 2;
        background: $gray5;
    }
}

.header-cell {
    background: none;
}

.config-icon {
    position: absolute;
    height: 100%;
    right: 24px;
    z-index: 2;
}