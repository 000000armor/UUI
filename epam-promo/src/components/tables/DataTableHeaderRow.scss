@use '../../../assets/styles/index' as *;

.root {
    --uui-dt-row-background: #{$gray5};
    --uui-dt-row-border-color: #{$gray30};
    --uui-dt-row-border-width: 1px;
}

.header-cell {
    background: none;
}

.config-icon {
    position: absolute;
    height: 100%;
    right: 24px;
    z-index: 2;
}
