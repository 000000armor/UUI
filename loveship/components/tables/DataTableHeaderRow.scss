@use '../../assets/styles/scss/index'  as *;

.root {
    background: $night50;
    border-bottom: 1px solid $night300;

    &:global(.uui-table-header-row .uui-table-fixed-section-left),
    &:global(.uui-table-header-row .uui-table-fixed-section-right) {
        z-index: 2;
        background: $night50;
    }
}

.header-cell {
    background: none;
}

.config-icon {
    position: absolute;
    right: 24px;
    height: 100%;
}