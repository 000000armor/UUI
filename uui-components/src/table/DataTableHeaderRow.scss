.root {
    isolation: isolate;
    z-index: 10;
}