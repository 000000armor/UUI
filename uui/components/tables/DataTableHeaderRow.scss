@use  '../../assets/styles/index' as *;

.header-cell {
    background: none;
}

.config-icon {
    position: absolute;
    height: 100%;
    right: 24px;
    z-index: 2;
}