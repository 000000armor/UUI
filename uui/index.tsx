import './assets/styles/font-faces.scss';
export * from './icons';
export * from './components';
// export * from './i18n';
export * from '@epam/uui-core';