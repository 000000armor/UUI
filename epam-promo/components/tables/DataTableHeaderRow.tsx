import * as React from 'react';
import { DataTableHeaderRow as uuiDataTableHeaderRow } from '@epam/uui-components';
import { DataTableHeaderRowProps, withMods } from '@epam/uui';
import { IconButton, DataTableHeaderCell, DataTableHeaderRowMods } from '../';
import * as css from './DataTableHeaderRow.scss';
import { ReactComponent as MoreIcon } from '@epam/assets/icons/common/action-settings-18.svg';

export const DataTableHeaderRow = withMods<DataTableHeaderRowProps, DataTableHeaderRowMods>(
    uuiDataTableHeaderRow,
    mods => [css.root],
    mods => ({
        renderCell: props => <DataTableHeaderCell
            key={ props.column.key + (props.value.columnsConfig?.[props.column.key]?.order || '')  }
            { ...props }
            size={ mods.size }
            textCase={ mods.textCase || 'normal' }
        />,
        renderConfigButton: () => <IconButton
            key='configuration'
            onClick={ mods.onConfigButtonClick }
            cx={ css.configIcon }
            color='gray60'
            icon={ MoreIcon }
        />,
    }));