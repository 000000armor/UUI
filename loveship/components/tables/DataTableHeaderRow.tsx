import React from 'react';
import { DataTableHeaderRow as uuiDataTableHeaderRow } from '@epam/uui-components';
import { DataTableHeaderRowProps, withMods } from '@epam/uui';
import { ReactComponent as GearIcon } from './../icons/settings-18.svg';
import * as css from './DataTableHeaderRow.scss';
import { DataTableHeaderCell } from './DataTableHeaderCell';
import { DataTableHeaderRowMods } from './types';
import { LinkButton } from '../buttons';

export const DataTableHeaderRow = withMods<DataTableHeaderRowProps, DataTableHeaderRowMods>(
    uuiDataTableHeaderRow,
    mods => [css.root],
    mods => ({
        renderCell: props => <DataTableHeaderCell
            key={ props.column.key + (props.value.columnsConfig?.[props.column.key]?.order || '')  }
            { ...props }
            size={ mods.size }
            textCase={ mods.textCase || 'normal' }
        />,
        renderConfigButton: () => <LinkButton
            key='configuration'
            onClick={ mods.onConfigButtonClick }
            cx={ css.configIcon }
            size='30'
            color='night600'
            icon={ GearIcon }
        />,
    }));