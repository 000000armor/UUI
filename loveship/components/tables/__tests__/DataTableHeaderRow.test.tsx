import React from "react";
import renderer from "react-test-renderer";
import {demoColumns} from "./dataMocks";
import {DataSourceState} from '@epam/uui';
import {DataTableHeaderRow} from "../DataTableHeaderRow";

jest.mock("react-dom", () => ({
    findDOMNode: jest.fn(),
}));

const dataSourceState: DataSourceState = {
    sorting: [{ field: 'name', direction: "asc" }],
};

describe("DataTableHeaderRow", () => {
    it("should render with default props", () => {
        const tree = renderer
            .create(<DataTableHeaderRow
                value={ dataSourceState }
                onValueChange={ jest.fn() }
                columns={ demoColumns }
            />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("should be rendered correctly with extra props", () => {
        const tree = renderer
            .create(<DataTableHeaderRow
                value={ dataSourceState }
                onValueChange={ jest.fn() }
                columns={ demoColumns }
                size='24'
                textCase='upper'
                allowColumnsResizing={ true }
                allowColumnsReordering={ true }
                selectAll={ { value: false, onValueChange: jest.fn() } }
            />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});