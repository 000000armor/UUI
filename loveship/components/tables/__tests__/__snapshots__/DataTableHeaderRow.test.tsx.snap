// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataTableHeaderRow should be rendered correctly with extra props 1`] = `
<div
  className="container uui-table-row-container root root uui-table-header-row container align-items-center"
  role="row"
  style={
    Object {
      "columnGap": undefined,
      "minWidth": "563px",
      "rowGap": undefined,
    }
  }
>
  <div
    className="section uui-table-fixed-section fixedColumnsSectionLeft uui-table-fixed-section-left"
    style={
      Object {
        "--uui-dt-cell-border-width": "1px",
        "flex": "0 0 96px",
        "minWidth": "96px",
      }
    }
  >
    <div
      aria-sort="none"
      className="-clickable cell size-24 padding-left-24 resizable"
      onClick={[Function]}
      onPointerEnter={[Function]}
      onPointerLeave={[Function]}
      onPointerMove={[Function]}
      onPointerUp={[Function]}
      onTouchStart={[Function]}
      role="columnheader"
      style={
        Object {
          "alignSelf": undefined,
          "flexBasis": "96px",
          "flexGrow": 0,
          "flexShrink": undefined,
          "minWidth": "96px",
          "textAlign": undefined,
        }
      }
    >
      <label
        className="container uui-checkbox-container root size-12 mode-form uui-theme-loveship checkbox -clickable"
        tabIndex={-1}
      >
        <div
          className="uui-checkbox"
        >
          <input
            checked={false}
            onChange={[Function]}
            tabIndex={0}
            type="checkbox"
          />
        </div>
      </label>
      <div
        className="tooltipWrapper"
      >
        <div
          className="iconCell uui-table-header-caption"
        >
          <div
            className="uui-font-sans uui-text-night700 root line-height-18 font-size-14 v-padding-9 uui-theme-loveship caption upperCase container"
          >
            ID
          </div>
          <button
            className="container uui-button-box uui-enabled root size-24 uui-has-left-icon uui-no-right-icon uui-theme-loveship icon sortIcon"
            onClick={[Function]}
            tabIndex={-1}
            type="button"
          >
            <div
              className="container uui-icon uui-enabled"
              style={Object {}}
            >
              <svg
                className=""
              />
            </div>
          </button>
        </div>
      </div>
      <div
        className="resizeMark -draggable"
        onMouseDown={[Function]}
      />
    </div>
    <div
      className="uui-scroll-shadow-right"
    />
  </div>
  <div
    className="section scrollingSection uui-table-scrolling-section"
    style={
      Object {
        "--uui-dt-cell-border-width": "1px",
        "flex": "2 0 467px",
        "minWidth": "467px",
      }
    }
  >
    <div
      aria-sort="ascending"
      className="-clickable cell size-24 resizable draggable"
      onClick={[Function]}
      onPointerDown={[Function]}
      onPointerEnter={[Function]}
      onPointerLeave={[Function]}
      onPointerMove={[Function]}
      onPointerUp={[Function]}
      onTouchStart={[Function]}
      role="columnheader"
      style={
        Object {
          "alignSelf": undefined,
          "flexBasis": "200px",
          "flexGrow": 1,
          "flexShrink": undefined,
          "minWidth": "200px",
          "textAlign": undefined,
        }
      }
    >
      <div
        className="tooltipWrapper"
      >
        <div
          className="iconCell uui-table-header-caption"
        >
          <div
            className="uui-font-sans uui-text-night700 root line-height-18 font-size-14 v-padding-9 uui-theme-loveship caption upperCase container"
          >
            name
          </div>
          <button
            className="container uui-button-box uui-enabled root size-24 uui-has-left-icon uui-no-right-icon uui-theme-loveship icon sortIcon sortIconActive"
            onClick={[Function]}
            tabIndex={-1}
            type="button"
          >
            <div
              className="container uui-icon uui-enabled"
              style={Object {}}
            >
              <svg
                className=""
              />
            </div>
          </button>
        </div>
      </div>
      <div
        className="resizeMark -draggable"
        onMouseDown={[Function]}
      />
    </div>
    <div
      aria-sort="none"
      className="-clickable cell size-24 resizable draggable"
      onClick={[Function]}
      onPointerDown={[Function]}
      onPointerEnter={[Function]}
      onPointerLeave={[Function]}
      onPointerMove={[Function]}
      onPointerUp={[Function]}
      onTouchStart={[Function]}
      role="columnheader"
      style={
        Object {
          "alignSelf": undefined,
          "flexBasis": "150px",
          "flexGrow": undefined,
          "flexShrink": undefined,
          "minWidth": "150px",
          "textAlign": undefined,
        }
      }
    >
      <div
        className="tooltipWrapper"
      >
        <div
          className="iconCell uui-table-header-caption"
        >
          <div
            className="uui-font-sans uui-text-night700 root line-height-18 font-size-14 v-padding-9 uui-theme-loveship caption upperCase container"
          >
            Level
          </div>
          <button
            className="container uui-button-box uui-enabled root size-24 uui-has-left-icon uui-no-right-icon uui-theme-loveship icon sortIcon"
            onClick={[Function]}
            tabIndex={-1}
            type="button"
          >
            <div
              className="container uui-icon uui-enabled"
              style={Object {}}
            >
              <svg
                className=""
              />
            </div>
          </button>
        </div>
      </div>
      <div
        className="resizeMark -draggable"
        onMouseDown={[Function]}
      />
    </div>
    <div
      aria-sort="none"
      className="-clickable cell size-24 padding-right-24 resizable draggable"
      onClick={[Function]}
      onPointerDown={[Function]}
      onPointerEnter={[Function]}
      onPointerLeave={[Function]}
      onPointerMove={[Function]}
      onPointerUp={[Function]}
      onTouchStart={[Function]}
      role="columnheader"
      style={
        Object {
          "alignSelf": undefined,
          "flexBasis": "120px",
          "flexGrow": 1,
          "flexShrink": 0,
          "minWidth": "120px",
          "textAlign": undefined,
        }
      }
    >
      <div
        className="tooltipWrapper"
      >
        <div
          className="iconCell uui-table-header-caption"
        >
          <div
            className="uui-font-sans uui-text-night700 root line-height-18 font-size-14 v-padding-9 uui-theme-loveship caption upperCase container"
          >
            DATE
          </div>
          <button
            className="container uui-button-box uui-enabled root size-24 uui-has-left-icon uui-no-right-icon uui-theme-loveship icon sortIcon"
            onClick={[Function]}
            tabIndex={-1}
            type="button"
          >
            <div
              className="container uui-icon uui-enabled"
              style={Object {}}
            >
              <svg
                className=""
              />
            </div>
          </button>
        </div>
      </div>
      <div
        className="resizeMark -draggable"
        onMouseDown={[Function]}
      />
    </div>
  </div>
</div>
`;

exports[`DataTableHeaderRow should render with default props 1`] = `
<div
  className="container uui-table-row-container root root uui-table-header-row container align-items-center"
  role="row"
  style={
    Object {
      "columnGap": undefined,
      "minWidth": "563px",
      "rowGap": undefined,
    }
  }
>
  <div
    className="section uui-table-fixed-section fixedColumnsSectionLeft uui-table-fixed-section-left"
    style={
      Object {
        "--uui-dt-cell-border-width": "1px",
        "flex": "0 0 96px",
        "minWidth": "96px",
      }
    }
  >
    <div
      aria-sort="none"
      className="-clickable cell size-36 padding-left-24"
      onClick={[Function]}
      role="columnheader"
      style={
        Object {
          "alignSelf": undefined,
          "flexBasis": "96px",
          "flexGrow": 0,
          "flexShrink": undefined,
          "minWidth": "96px",
          "textAlign": undefined,
        }
      }
    >
      <div
        className="tooltipWrapper"
      >
        <div
          className="iconCell uui-table-header-caption"
        >
          <div
            className="uui-font-sans uui-text-night700 root line-height-18 font-size-14 v-padding-9 uui-theme-loveship caption font-size-14 container"
          >
            ID
          </div>
          <button
            className="container uui-button-box uui-enabled root size-24 uui-has-left-icon uui-no-right-icon uui-theme-loveship icon sortIcon"
            onClick={[Function]}
            tabIndex={-1}
            type="button"
          >
            <div
              className="container uui-icon uui-enabled"
              style={Object {}}
            >
              <svg
                className=""
              />
            </div>
          </button>
        </div>
      </div>
    </div>
    <div
      className="uui-scroll-shadow-right"
    />
  </div>
  <div
    className="section scrollingSection uui-table-scrolling-section"
    style={
      Object {
        "--uui-dt-cell-border-width": "1px",
        "flex": "2 0 467px",
        "minWidth": "467px",
      }
    }
  >
    <div
      aria-sort="ascending"
      className="-clickable cell size-36"
      onClick={[Function]}
      role="columnheader"
      style={
        Object {
          "alignSelf": undefined,
          "flexBasis": "200px",
          "flexGrow": 1,
          "flexShrink": undefined,
          "minWidth": "200px",
          "textAlign": undefined,
        }
      }
    >
      <div
        className="tooltipWrapper"
      >
        <div
          className="iconCell uui-table-header-caption"
        >
          <div
            className="uui-font-sans uui-text-night700 root line-height-18 font-size-14 v-padding-9 uui-theme-loveship caption font-size-14 container"
          >
            name
          </div>
          <button
            className="container uui-button-box uui-enabled root size-24 uui-has-left-icon uui-no-right-icon uui-theme-loveship icon sortIcon sortIconActive"
            onClick={[Function]}
            tabIndex={-1}
            type="button"
          >
            <div
              className="container uui-icon uui-enabled"
              style={Object {}}
            >
              <svg
                className=""
              />
            </div>
          </button>
        </div>
      </div>
    </div>
    <div
      aria-sort="none"
      className="-clickable cell size-36"
      onClick={[Function]}
      role="columnheader"
      style={
        Object {
          "alignSelf": undefined,
          "flexBasis": "150px",
          "flexGrow": undefined,
          "flexShrink": undefined,
          "minWidth": "150px",
          "textAlign": undefined,
        }
      }
    >
      <div
        className="tooltipWrapper"
      >
        <div
          className="iconCell uui-table-header-caption"
        >
          <div
            className="uui-font-sans uui-text-night700 root line-height-18 font-size-14 v-padding-9 uui-theme-loveship caption font-size-14 container"
          >
            Level
          </div>
          <button
            className="container uui-button-box uui-enabled root size-24 uui-has-left-icon uui-no-right-icon uui-theme-loveship icon sortIcon"
            onClick={[Function]}
            tabIndex={-1}
            type="button"
          >
            <div
              className="container uui-icon uui-enabled"
              style={Object {}}
            >
              <svg
                className=""
              />
            </div>
          </button>
        </div>
      </div>
    </div>
    <div
      aria-sort="none"
      className="-clickable cell size-36 padding-right-24"
      onClick={[Function]}
      role="columnheader"
      style={
        Object {
          "alignSelf": undefined,
          "flexBasis": "120px",
          "flexGrow": 1,
          "flexShrink": 0,
          "minWidth": "120px",
          "textAlign": undefined,
        }
      }
    >
      <div
        className="tooltipWrapper"
      >
        <div
          className="iconCell uui-table-header-caption"
        >
          <div
            className="uui-font-sans uui-text-night700 root line-height-18 font-size-14 v-padding-9 uui-theme-loveship caption font-size-14 container"
          >
            DATE
          </div>
          <button
            className="container uui-button-box uui-enabled root size-24 uui-has-left-icon uui-no-right-icon uui-theme-loveship icon sortIcon"
            onClick={[Function]}
            tabIndex={-1}
            type="button"
          >
            <div
              className="container uui-icon uui-enabled"
              style={Object {}}
            >
              <svg
                className=""
              />
            </div>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
